### PR TITLE
Overhaul of plugin template for modern CMake build system and GitHub Actions

### PR DESCRIPTION
**Pull request by PatTheMav**
### Description
Updates the entire template in anticipation of https://github.com/obsproject/obs-studio/pull/4560:

* Use updated external plugin helper
* Use the modern CMake package exports introduced by the overhaul in the main repo
* Generates plugin bundles on macOS, including code signing and fixing library rpath's
* Uses user library install path on macOS, removing the need for elevated privileges during installation
* Uses Powershell scripts on Windows
* Uses identical script steps on all three Ones
* Uses included subscripts to allow local builds and CI to use the same scripts
* Replaces Azure Pipelines with GitHub Actions (just like the main repo did)
* Notarizes on macOS by default if necessary conditions are met (release tag and credentials set)
* Uses Xcnotary to run notarization (just like main repo)

**Note**: This PR will remain in draft stage until the main PR mentioned above has been merged, as it requires the updated OBS branch.

### Motivation and Context
Allows creation and maintenance of obs plugins with less effort as provided scripts handle finding libraries, packaging, mixups and other necessary tasks and reduces number of workarounds. Also does _not_ require developers to specify directories containing necessary dependencies anymore (as the updated OBS build scripts properly export `libobs`, `obs-frontend-api` as well as `w32-pthreads` (on Windows)) and link to those by default.

### How Has This Been Tested?
* Ran main build script as well as subscripts on macOS, Linux (ElementaryOS 6) and Windows 10

### Types of changes
- New feature (non-breaking change which adds functionality)
- Code cleanup (non-breaking change which makes code smaller or more readable)
- Breaking change (fix or feature that would cause existing functionality to change)
- Documentation (a change to documentation pages)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [ ] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+        with:
+          path: |
+            ${{ github.workspace }}/obs-studio/build/libobs
+            ${{ github.workspace }}/obs-studio/build/UI/obs-frontend-api
+            ${{ github.workspace }}/obs-studio/build/deps/w32-pthreads
+          key: ${{ runner.os }}-pr-${{ env.CACHE_NAME }}-${{ env.OBS_VERSION }}
+
+      - name: 'Build libobs and obs-frontend-api'
+        working-directory: 'obs-studio'
+        run: CI/windows/02_build_obs-libs.ps1 -BuildArch "${{ matrix.arch }}-bit"
```
Comment (norihiro) [inline .github/workflows/main.yml:274]:
```suggestion
        run: CI/windows/02_build_obs_libs.ps1 -BuildArch "${{ matrix.arch }}-bit"
```

Comment (PatTheMav) [reply .github/workflows/main.yml]: Good catch, fixed.

**Comment by RytoEX** [comment]
Does this require obsproject/obs-studio#4560 to be merged first, or can this be merged first?

**Comment by tytan652** [comment]
> Does this require [obsproject/obs-studio#4560](https://github.com/obsproject/obs-studio/pull/4560) to be merged first, or can this be merged first?

To work in-tree at least. I think.

**Comment by PatTheMav** [comment]
Most of the rework's changes are implemented in https://github.com/PatTheMav/obs-plugintemplate/blob/template-overhaul/external/ObsPluginHelpers.cmake without any dependencies on the cmake rework.

When `libobs` and `ons-frontend-api` are specified manually (as seems common right now) it should work out of the box and still generate "modern" plugin packages.

Once the cmake-rework merged it would find the libraries automatically.

**Pull request merged**